# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/client/fronts/watcher.js
# Commit: 895cf08432c4
# Full Hash: 895cf08432c40fe371ff83e846cb161430c6be4b
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-10-27 04:41:26
# Regressor Bug: 1633712
# File Overlap Count: 1
# Description:
#   Bug 1633712 - [devtools] Create WorkerTargets as soon as possible. r=ochameau,devtools-backward-compat-reviewers.
#   
#   This patch adds support for dedicated worker targets in the Watcher actor.
#   Shared and Service workers are not handled yet.
#   
# ==============================================================================

diff -r 2ce79a9ed2d5 -r 895cf08432c4 devtools/client/fronts/watcher.js
--- a/devtools/client/fronts/watcher.js	Mon Oct 26 12:52:26 2020 +0000
+++ b/devtools/client/fronts/watcher.js	Mon Oct 26 12:53:43 2020 +0000
@@ -21,6 +21,12 @@
   "devtools/client/fronts/targets/content-process",
   true
 );
+loader.lazyRequireGetter(
+  this,
+  "WorkerTargetFront",
+  "devtools/client/fronts/targets/worker",
+  true
+);
 
 class WatcherFront extends FrontClassWithSpec(watcherSpec) {
   constructor(client, targetFront, parentFront) {
@@ -43,6 +49,8 @@
     let front;
     if (form.actor.includes("/contentProcessTarget")) {
       front = new ContentProcessTargetFront(this.conn, null, this);
+    } else if (form.actor.includes("/workerTarget")) {
+      front = new WorkerTargetFront(this.conn, null, this);
     } else {
       front = new BrowsingContextTargetFront(this.conn, null, this);
     }